src/openloco/windows/music_selection.cpp

```diff
@@ -0,0 +1,236 @@
+        if (window != nullptr)
+            return window;
+
+        window = WindowManager::createWindow(
+            WindowType::musicSelection,
+            { ww, wh },
+            0,
+            &_events);
+
+        // TODO: only needs to be called once.
+        init_events();
+
+        window->widgets = _widgets;
+        window->enabled_widgets = 1 << widx::close;
+        window->init_scroll_widgets();
+
+        auto interface = objectmgr::get<interface_skin_object>();
+        window->colours[0] = interface->colour_0B;
+        window->colours[1] = interface->colour_10;
+
+        window->row_count = numTracks;
+        window->row_hover = -1;
+
+        return window;
+    }
+
+    // 0x004C165D
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        // Draw widgets.
+        window->draw(dpi);
+    }
+
+    // 0x004C1663
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+    {
+        auto shade = colour::get_shade(window->colours[1], 4);
+        gfx::clear_single(*dpi, shade);
+
+        static const string_id music_ids_to_string_id[] = {
+            string_ids::music_chuggin_along,
+            string_ids::music_long_dusty_road,
+            string_ids::music_flying_high,
+            string_ids::music_gettin_on_the_gas,
+            string_ids::music_jumpin_the_rails,
+            string_ids::music_smooth_running,
+            string_ids::music_traffic_jam,
+            string_ids::music_never_stop_til_you_get_there,
+            string_ids::music_soaring_away,
+            string_ids::music_techno_torture,
+            string_ids::music_everlasting_high_rise,
+            string_ids::music_solace,
+            string_ids::music_chrysanthemum,
+            string_ids::music_eugenia,
+            string_ids::music_the_ragtime_dance,
+            string_ids::music_easy_winners,
+            string_ids::music_setting_off,
+            string_ids::music_a_travellers_seranade,
+            string_ids::music_latino_trip,
+            string_ids::music_a_good_head_of_steam,
+            string_ids::music_hop_to_the_bop,
+            string_ids::music_the_city_lights,
+            string_ids::music_steamin_down_town,
+            string_ids::music_bright_expectations,
+            string_ids::music_mo_station,
+            string_ids::music_far_out,
+            string_ids::music_running_on_time,
+            string_ids::music_get_me_to_gladstone_bay,
+            string_ids::music_sandy_track_blues,
+        };
+
+        auto config = config::get();
+
+        uint16_t y = 0;
+        for (uint16_t i = 0; i < window->row_count; i++)
+        {
+            string_id text_colour_id = string_ids::white_stringid2;
+
+            // Draw hovered track
+            if (i == window->row_hover)
+            {
+                gfx::draw_rect(dpi, 0, y, 800, rowHeight - 1, 0x2000030);
```

Comment:
```suggestion
                gfx::draw_rect(dpi, 0, y, 800, rowHeight, 0x2000030);
```

(the original code used the top-left corner of pixels. Techincally it also should be 801 instead of 800, but 800 is a nicer number)

Reply:
We don't have to necessary adhere to the original code, imo. I also contemplated changing 800 to the widget width — no need to draw more than is visible, right?